var/spack/repos/builtin/packages/clingo-bootstrap/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.pkg.builtin.clingo import Clingo
+
+import spack.compilers
+
+
+class ClingoBootstrap(Clingo):
+    """Clingo with some options used for bootstrapping"""
+    variant('build_type', default='Release', values=('Release',),
+            description='CMake build type')
+    depends_on('cmake@3.16.0:', type='build')
+
+    # On Linux we bootstrap with GCC
+    for compiler_spec in [
+        c for c in spack.compilers.supported_compilers()
+        if c != 'gcc'
+    ]:
+        conflicts('%{0}'.format(compiler_spec), when='platform=linux',
+                  msg='GCC is required to bootstrap clingo on Linux')
+    conflicts('%gcc@:5.99.99', when='platform=linux',
+              msg='C++14 support is required to bootstrap clingo on Linux')
+
+    # On Darwin we bootstrap with Apple Clang
+    for compiler_spec in [
+        c for c in spack.compilers.supported_compilers()
+        if c != 'apple-clang'
+    ]:
+        conflicts('%{0}'.format(compiler_spec), when='platform=darwin',
+                  msg='Apple-clang is required to bootstrap clingo on MacOS')
+
+    # Clingo needs the Python module to be usable by Spack
+    conflicts('~python', msg='Python support is required to bootstrap Spack')
+
+    @property
+    def cmake_python_hints(self):
+        """Return standard CMake defines to ensure that the
+        current spec is the one found by CMake find_package(Python, ...)
+        """
+        return [
+            '-DPython_EXECUTABLE={0}'.format(str(self.spec['python'].command))
+        ]
```

Comment:
Is there a reason this isn't in the `clingo` package?

Reply:
I put this here because it imposes a dependency on recent CMake ( >= 3.16.0). So I thought we don't want to impose this to people building `clingo` with a Spack built `python`.

Reply:
@becker33 Is there anything left to do here?

Reply:
I think we should make this argument conditional on the cmake version, and always emit it for sufficiently new cmake. That will make the most likely case (new-ish cmake) more robust for external python.

Reply:
But if that's controversial, I'm okay with punting that to the future and merging this as-is.

Reply:
> But if that's controversial, I'm okay with punting that to the future and merging this as-is.

No, that's definitely a good idea, I didn't think about having it conditional.

Reply:
@becker33 Done